x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/action/cohere/CohereActionCreatorTests.java

```diff
@@ -67,9 +68,9 @@ public void shutdown() throws IOException {
     }
 
     public void testCreate_CohereEmbeddingsModel() throws IOException {
-        var senderFactory = new HttpRequestSenderFactory(threadPool, clientManager, mockClusterServiceEmpty(), Settings.EMPTY);
+        var senderFactory = HttpRequestSenderTests.createSenderFactory(threadPool, clientManager);
 
-        try (var sender = senderFactory.createSender("test_service")) {
+        try (var sender = senderFactory.createSender("test_service", new SingleRequestManager.Factory())) {
```

Comment:
Ah, I see this is to support batching in the future. I guess I'm still a bit confused about why we would want to have different classes for a SingleRequest and MultipleRequests, rather than just sending a list with only one request in it to a RequestManager.